### PR TITLE
Ensure that tokens are re-fetched to the position in the reset point in SyntaxParser.Reset

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes [AB#1179569](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1179569).

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review a fix for a Watson bug.

File: src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs [hunk 1]
```diff
@@ -150,8 +150,19 @@ protected ResetPoint GetResetPoint()
...
+            {
+                // Re-fetch tokens to the position in the reset point
+                PeekToken(offset - _tokenOffset);
+
+                // Re-calculate new offset in case tokens got shifted to the left while we were peeking. 
+                offset = point.Position - _firstToken;
+            }
+
+            _mode = point.Mode;
             Debug.Assert(offset >= 0 && offset < _tokenCount);
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs:166]: nit: consider adding a similar assertion in `GetResetPoint`

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review a fix for a Watson bug, need a second sign-off.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-ide, @dotnet/roslyn-infrastructure Are the failures of integration tests expected?

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review a fix for a Watson bug, need a second sign-off.

**Comment by AlekseyTs** [comment]
@cston Please review.

**Review by jaredpar — APPROVED** [review]

**Pull request merged**